public/app/user-service-registry/user-service-registry.controller.js

```diff
@@ -21,86 +21,411 @@
-                    params = $scope.params.split(", ");
-                    if (params.length == 1) {
-                        params = params[0].split(",");
-                    }
+
+            var resourceMetadata = {
+                "name":$scope.serviceName,
+                "description":$scope.serviceDescription,
+                "url":$scope.serviceUrl,
+                "method":$scope.method
+            };
+
+            var data = {
+                "resourceMetadata" : resourceMetadata,
+                "inputs" : $scope.inputs,
+                "outputs" : $scope.outputs
+            };
+            var job = {
+                "apiKey": "my-api-key-38n987",
+                "jobType" : {
+                    "type": "register-service",
+                    "data" : data
                 }
-                var data = {
-                    "name":$scope.serviceName,
-                    "description":$scope.serviceDescription,
-                    "url":$scope.serviceUrl,
-                    "method":$scope.method,
-                    "params":params,
-                    "responseMimeType":$scope.responseType
-                };
-                $http.post(
-                    "/proxy?url=" + result.data.address + "/servicecontroller/registerService",
-                    data,
-                    {
-                        headers: {
-                            "Content-Type": "application/json"
-                        }
-                    }
-                ).then(function successCallback( html ) {
-                    $scope.resourceId = html.data.resourceId;
-                }, function errorCallback(response){
-                    console.log("user-service-registry.controller fail");
-                    toaster.pop('error', "Error", "There was an issue with your request.");
-                });
+            };
+
+            var fd = new FormData();
+            fd.append( 'body', angular.toJson(job) );
+
+
+
 
+            var request = $http({
+                method: "POST",
+                url: '/proxy?url=pz-gateway.stage.geointservices.io/job',
+                data :fd,
+                headers: {"Content-Type": undefined}
+            }).then(function successCallback( html ) {
+               $scope.jobId = html.data.jobId;
+                getRegisterResult($scope.jobId)
+
+
+            }, function errorCallback(response){
+                console.log("user-service-registry.controller fail");
+                toaster.pop('error', "Error", "There was an issue with your request.");
             });
+
         };
 
+        function createExecuteInputMap(inputs) {
+            //executeInputMap
+            var i = 0;
+            for (i = 0; i < inputs.length; i++) {
+                switch (inputs[i].dataType.type) {
+                    // TODO: Looks like this can be simplified because body, raster and text all do the same thing
+                    case "body":
```

Comment:
This switch can be simplified using fall-through because 3 out of 4 results do the same thing.